### PR TITLE
feat: add API calls to get data needed by the bulk email task manager section

### PR DESCRIPTION
**Pull request by justinhynes**
[MICROBA-1621]

* Add support to the Instructor Dashboard `list_instructor_tasks`, `list_email_content`, and `list_background_email_tasks`  REST API endpoints of edx-platform.

### CODE REVIEW

File: src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailContentHistory.jsx [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
 
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { getSentEmailHistory } from './api';
 
 export default function BulkEmailContentHistory() {
+  const { courseId } = useParams();
+
+  const [emailHistoryData, setEmailHistoryData] = useState(); // eslint-disable-line no-unused-vars
```
Comment (justinhynes) [inline src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailContentHistory.jsx:10]: While `emailHistoryData` isn't being used _now_, we will use it in a follow-up to this work. Disabling the es-lint check is just temporary and will be removed.

File: src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailPendingTasks.jsx [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
 
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { getInstructorTasks } from './api';
 
 export default function BulkEmailPendingTasks() {
+  const { courseId } = useParams();
+
+  const [instructorTaskData, setInstructorTaskData] = useState(); // eslint-disable-line no-unused-vars
```
Comment (justinhynes) [inline src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailPendingTasks.jsx:10]: See comment in BulkEmailContentHistory.jsx

File: src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailTaskHistory.jsx [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
 
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { getEmailTaskHistory } from './api';
 
 export default function BulkEmailTaskHistory() {
+  const { courseId } = useParams();
+
+  const [emailTaskHistoryData, setEmailTaskHistoryData] = useState(); // eslint-disable-line no-unused-vars
```
Comment (justinhynes) [inline src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailTaskHistory.jsx:10]: See comment in BulkEmailContentHistory.jsx

**Review by Tj-Tracy — APPROVED** [review]
LGTM 👍 I like the `api` setup

File: src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailContentHistory.jsx [hunk 4]
```diff
@@ -1,8 +1,23 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
 
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { getSentEmailHistory } from './api';
```
Comment (hurtstotouchfire) [inline src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailContentHistory.jsx:5]: What is `./api`? The name seems very generic, is it the API of a library provided by `frontend-app-communications`?

Comment (justinhynes) [reply src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailContentHistory.jsx:5]: That is referencing a local `api.js` file for the `bulk-email-task-manager` component itself. The pattern we follow, as I understand it, is that each component can have their own `api.js` that contains whatever REST API functions needed that the component will rely on.

**Pull request merged**